ledger/apply/asset.go

```diff
@@ -330,6 +330,10 @@ func AssetTransfer(ct transactions.AssetTransferTxnFields, header transactions.H
 		// and putIn will short circuit (so bypassFreeze doesn't matter)
 		_, bypassFreeze := dst.AssetParams[ct.XferAsset]
 
+		// Add the close amount to ApplyData.
+		closeAmount := sndHolding.Amount
+		ad.AssetClosingAmount = closeAmount
+
```

Comment:
This part has to be enclosed in a protocol upgrade parameter condition. specifically, you'll need to PR https://github.com/algorandfoundation/specs/blob/3a83c4c743f8b17adfd73944b4319c25722a6782/dev/ledger.md and have it merged. Once it's merged, you'll have a commit hash that you can use for the consensus protocol string.

Reply:
https://github.com/algorandfoundation/specs/pull/34